packages/flutter/lib/src/material/radio.dart

```diff
@@ -21,11 +21,49 @@ const double _kInnerRadius = 4.5;
 /// be selected. The values are of type `T`, the type parameter of the [Radio]
 /// class. Enums are commonly used for this purpose.
 ///
-/// The radio button itself does not maintain any state. Instead, when the state
-/// of the radio button changes, the widget calls the [onChanged] callback.
-/// Most widget that use a radio button will listen for the [onChanged]
-/// callback and rebuild the radio button with a new [groupValue] to update the
-/// visual appearance of the radio button.
+/// The radio button itself does not maintain any state. Instead, selecting the
+/// radio invokes the [onChanged] callback, passing [value] as a parameter. If
+/// [groupValue] and [value] match, this radio will be selected. Most widgets
+/// will listen to [onChanged] with setState to update the radio's visual
+/// appearance.
+///
+/// {@tool snippet --template=stateful_widget_scaffold}
+///
+/// Here is an example of Radio widgets wrapped in ListTiles.
```

Comment:
If this is similar to what you get with RadioListTile, you could mention as much here.

It would be good to provide one or two sentences that explain what's going on. Something like: The currently selected character - the radio buttons' "group value" - is maintained by the example's [State]. When one of the radio buttons is pressed, the example's state is updated with [State.setState], which causes the buttons to be rebuilt with a new group value.